src/TestSuite/TestCase.php

```diff
@@ -203,13 +203,37 @@ public function withErrorReporting(int $errorLevel, callable $callable): void
      */
     public function deprecated(callable $callable): void
     {
-        $errorLevel = error_reporting();
-        error_reporting($errorLevel & ~E_USER_DEPRECATED);
+        $duplicate = Configure::read('Error.enableDuplicatedDeprecations');
```

Comment:
You shouldn't need to make this variable or restore the value at the end of the test method. The `tearDown()` of `TestCase` will restore all of `Configure` anyways.

Reply:
not in this case
1)
```php

    public function testMrthod(): void
{
           $this->expectDeprecation();
           $this->expectDeprecationMessage('after deprecated')
           $this->deprecated(function () {
                     trigger_error('message within deprecated method', E_USER_DEPRECATED);
                     trigger_error('other message', E_USER_DEPRECATED);
            });

            trigger_error('after deprecated', E_USER_DEPRECATED);
}
```
we want to above testCase passes

```php
    public function testMrthod(): void
{
           // section 1
           $this->deprecated(function () {
                     // section 2
            });

            // section 3
}
```

by default in `section 1` and `section 3` deprecations will trigger unless if there was duplicate deprecation
(not included warnings from `deprecated()` 

in `section 2` we must trigger all deprecation even duplicated, ( but we should not include them into `checksum` )

`deprecated()`  just works like before this PR,
we dont want to change other sections 

so we have to back to previous Config after finishing the `deprecated()` block


2) by  default we prevent to show duplicated deprecation unless user want to use `deprecated()` 
so we did not change this Config before (in `setUp` method ), so we can not understand what was before to restore it in `tearDown()`

Reply:
I did not consider that scenario. Thank you for the great explanation :bow: